resources/aws/subnet.go

```diff
@@ -100,17 +103,23 @@ func (s *Subnet) CreateOrFail() error {
 }
 
 func (s *Subnet) Delete() error {
```

Comment:
IMO `Subnet` is a genetal logic provider. Thus `Delete` should expect an ID argument identifying the subnet to delete. This separates logic from data structures.